kentikapi/models/cloud_export.go

```diff
@@ -0,0 +1,243 @@
+package models
+
+// NewAWSCloudExport creates a new AWS CloudExport with all required fields set.
+func NewAWSCloudExport(obj CloudExportAWSRequiredFields) *CloudExport {
+	return &CloudExport{
+		Name:          obj.Name,
+		PlanID:        obj.PlanID,
+		CloudProvider: obj.CloudProvider,
+		AWSProperties: &AWSProperties{
+			Bucket: obj.AWSProperties.Bucket,
+		},
+	}
+}
+
+// NewAzureCloudExport creates a new AZURE CloudExport with all required fields set.
+func NewAzureCloudExport(obj CloudExportAzureRequiredFields) *CloudExport {
+	return &CloudExport{
+		Name:          obj.Name,
+		PlanID:        obj.PlanID,
+		CloudProvider: obj.CloudProvider,
+		AzureProperties: &AzureProperties{
+			Location:       obj.AzureProperties.Location,
+			ResourceGroup:  obj.AzureProperties.ResourceGroup,
+			StorageAccount: obj.AzureProperties.StorageAccount,
+			SubscriptionID: obj.AzureProperties.SubscriptionID,
+		},
+	}
+}
+
+// NewGCECloudExport creates a new GCE CloudExport with all required fields set.
+func NewGCECloudExport(obj CloudExportGCERequiredFields) *CloudExport {
+	return &CloudExport{
+		Name:          obj.Name,
+		PlanID:        obj.PlanID,
+		CloudProvider: obj.CloudProvider,
+		GCEProperties: &GCEProperties{
+			Project:      obj.GCEProperties.Project,
+			Subscription: obj.GCEProperties.Subscription,
+		},
+	}
+}
+
+// NewIBMCloudExport creates a new IBM CloudExport with all required fields set.
+func NewIBMCloudExport(obj CloudExportIBMRequiredFields) *CloudExport {
+	return &CloudExport{
+		Name:          obj.Name,
+		PlanID:        obj.PlanID,
+		CloudProvider: obj.CloudProvider,
+		IBMProperties: &IBMProperties{
+			Bucket: obj.IBMProperties.Bucket,
+		},
+	}
+}
+
+// GetAllCloudExportsResponse model.
+type GetAllCloudExportsResponse struct {
+	// CloudExports holds all cloud export tasks.
+	CloudExports []CloudExport
+	// InvalidCloudExportsCount is a number of invalid cloud export tasks.
+	InvalidCloudExportsCount uint32
+}
+
+// CloudExport defines a cloud export task.
+type CloudExport struct {
+	// Read-write properties
+
+	// Type of export task.
+	Type CloudExportType
+	// Enabled specifies whether this task is enabled and intended to run or disabled.
+	Enabled *bool
```

Comment:
why is this a pointer?

Reply:
In short: for consistency with how other booleans are handled.

Reply:
OK